core/templates/components/forms/custom-forms-directives/thumbnail-uploader.directive.ts

```diff
@@ -52,24 +53,32 @@ angular.module('oppia').directive('thumbnailUploader', [
         function($rootScope, $scope, $uibModal,
             AlertsService, ContextService, CsrfTokenService,
             ImageUploadHelperService) {
-          var placeholderImageUrl = '/icons/story-image-icon.png';
-          $scope.placeholderImageDataUrl = (
+          var placeholderImageDataUrl = (
             UrlInterpolationService.getStaticImageUrl(
-              placeholderImageUrl));
+              '/icons/story-image-icon.png'));
           var uploadedImage = null;
-          $scope.imageContainerStyle = {};
-          if ($scope.getFilename()) {
-            $scope.editableThumbnailDataUrl = (
-              ImageUploadHelperService
-                .getTrustedResourceUrlForThumbnailFilename(
-                  $scope.getFilename(),
-                  ContextService.getEntityType(),
-                  ContextService.getEntityId()));
-            uploadedImage = $scope.editableThumbnailDataUrl;
-            $scope.imageContainerStyle = {
-              background: $scope.getBgColor()
-            };
-          }
+          $scope.thumbnailIsLoading = false;
+          // $watch is required here to update the thumbnail image
+          // everytime the thumbnail filename changes (eg. draft is discarded).
+          // The trusted resource url for the thumbnail should not be directly
+          // bound to ngSrc because it can cause an infinite digest error.
+          // eslint-disable-next-line max-len
+          // https://github.com/angular/angular.js/blob/master/CHANGELOG.md#sce-
+          $scope.$watch('getFilename()', function(filename) {
```

Comment:
This worries me slightly because $watch is expensive. Do we do a similar thing anywhere else?